pandas/_libs/tslibs/conversion.pyx

```diff
@@ -32,6 +50,31 @@ cdef int64_t DAY_NS = 86400000000000LL
 
 UTC = pytz.UTC
 
+# ----------------------------------------------------------------------
+# Misc Helpers
+
+cdef inline bint _is_timestamp(datetime obj):
```

Comment:
rename this to make it clear

``is_instance_timestamp``

Reply:
if tihs is *onlyI used in that 1 place I would just in-line it directly.

Reply:
> if tihs is *onlyI used in that 1 place I would just in-line it directly.

Makes sense.

Just realized: recall recently there was an issue with `Timestamp` classmethods returning `Timestamp` instead of `cls` instances.  The conclusion was that this could not be fixed because `is_timestamp` would break on subclasses.  If this implementation is in fact equally performant, that would solve the problem.